packages/hash/api/src/graph/util.ts

```diff
@@ -218,11 +223,10 @@ export type EntityTypeCreatorParams = {
   }[];
   outgoingLinks?: {
     linkEntityType: EntityTypeWithMetadata;
-    destinationEntityTypes: (
-      | EntityTypeWithMetadata
-      // Some models may reference themselves. This marker is used to stop infinite loops during initialization by telling the initializer to use a self reference
-      | "SELF_REFERENCE"
-    )[];
+    destinationEntityTypes?: [
```

Comment:
Made this optional, the Type System doesn't allow an empty `oneOf`, it allows an _optional_ `oneOf`